include/ddc/kernels/splines/matrix.hpp

```diff
@@ -107,11 +173,23 @@ class Matrix
         return bx;
     }
 
+    /**
+     * @brief Get the size of the matrix (which is necessarily squared) in one of its dimensions.
+     *
+     * @return The size of the matrix in one of its dimensions.
+     */
     int get_size() const
     {
         return m_n;
     }
 
+    /**
+     * @brief Prints a Matrix in a std::ostream. It will segfault is the Matrix is on GPU.
```

Comment:
```suggestion
     * @brief Prints a Matrix in a std::ostream. It must not be called after `factorize`.
```

I would allow it to be called before factorize, implementations must provide a working `get_element` before `factorize`